docker_cli/image.py

```diff
@@ -0,0 +1,430 @@
+import io
+import os
+from shlex import split
+from subprocess import PIPE, STDOUT, CalledProcessError, run
+from sys import stdin
+from typing import (Any, List, Optional, Sequence, Type, TypeVar, Union,
+                    overload)
+
+from .exceptions import CommandNotFoundOnImageError
+
+
+class Image:
+    """A class which holds a reference to a given docker image and provides
+    an interface by which to interact with that image.
+
+    -   Capable of building docker images from dockerfiles or dockerfile-
+        formatted strings via the build classmethods.
+    -   Capable of acquiring information about the given image (see property
+        methods below).
+    -   Capable of running commands on containers built from the image using
+        the run method.
+    -   All docker build and run methods on the class can recive a file-like
+        object, (io.TextIOBase or some subclass of it,) and they will write the
+        outputs from docker into the file to enable logging.
+
+    NOTE: This class does not work correctly if the image built does not have
+        bash installed.
+    """
+    Self = TypeVar("Self", bound="Image")
+
+    def __init__(self, name_or_id: str):
+        """Initialize this Image. Connects the class with its ID.
+
+        Parameters
+        ----------
+        name_or_id : str
+            A name or ID by which to find this image using docker inspect.
+        """
+        self._id = get_image_id(name_or_id)
+
+    @overload
+    @classmethod
+    def build(
+        cls: Type[Self],
+        context: os.PathLike[str],
+        tag: str,
+        *,
+        dockerfile: os.PathLike[str],
+        output_file: Optional[io.TextIOBase],
+        network: str,
+        no_cache: bool
+    ):
+        """Build a Dockerfile at the given path with the given name, then
+        return the associated Image instance.
+
+        Parameters
+        ----------
+        context : str
+            The path to the file containing the Dockerfile
+        tag : str
+            A name for the image
+        dockerfile : os.PathLike, optional
+            The location of the Dockerfile to build, relative to the position
+            held at the context argument described above.
+        output_file : io.TextIOBase, optional
+            A file-like object that the stderr output of the docker build
```

Comment:
As currently implemented, it looks like this file is getting STDOUT messages, not STDERR.

```suggestion
            A file-like object that the stdout output of the docker build
```